src/Ext/Techno/Body.Update.cpp

```diff
@@ -65,6 +65,70 @@ void TechnoExt::ExtData::ApplyInterceptor()
 	}
 }
 
+//test
+void TechnoExt::ExtData::DepletedAmmoActions()
+{
+	auto const pThis = this->OwnerObject();
+	auto const pType = pThis->GetTechnoType();
+	bool const isUnit = pThis->WhatAmI() == AbstractType::Unit;
+
+	if (!isUnit && (pType->Ammo <= 0))
+		return;
```

Comment:
```suggestion
	if ((pThis->WhatAmI() != AbstractType::Unit) || (pType->Ammo > 0))
		return;
```

Not sure why there was an AND instead of OR.